cdap-app-fabric/src/main/java/io/cdap/cdap/gateway/handlers/SupportBundleHttpHandler.java

```diff
@@ -0,0 +1,610 @@
+                            latestProgramRunId.getApplication(),
+                            type,
+                            latestProgramRunId.getProgram(),
+                            latestProgramRunId.getRun());
+                    LoggingContext loggingContext =
+                        LoggingContextHelper.getLoggingContextWithRunId(latestProgramRunId, null);
+                    generateLogFile(
+                        loggingContext, appPath, latestProgramRunId.getRun() + "-log.txt");
+                  } catch (Exception e) {
+                    e.printStackTrace();
+                  }
+                }
+              });
+      thread.start();
+      threadList.add(thread);
+      addToStatus(statusJson, "runtimelog");
+    }
+    // Generate runtime info details file
+    final Thread thread =
+        new Thread(
+            new Runnable() {
+              @Override
+              public void run() {
+                try {
+                  JsonObject jsonObject = new JsonObject();
+                  jsonObject.addProperty("status", latestRunRecordDetail.getStatus().toString());
+                  jsonObject.addProperty("start", latestRunRecordDetail.getStartTs());
+                  jsonObject.addProperty("end", latestRunRecordDetail.getStopTs());
+                  jsonObject.addProperty(
+                      "profileName", latestRunRecordDetail.getProfileId().getProfile());
+                  jsonObject.addProperty(
+                      "runtimeArgs", latestRunRecordDetail.getProperties().get("runtimeArgs"));
+                  jsonObject.add("metrics", metrics);
+                  FileWriter file =
+                      new FileWriter(appPath + "/" + latestProgramRunId.getRun() + ".json");
+                  file.write(GSON.toJson(jsonObject));
+                  file.flush();
+                  file.close();
+                } catch (IOException e) {
+                  e.printStackTrace();
+                }
+              }
+            });
+    thread.start();
+    threadList.add(thread);
+    addToStatus(statusJson, "runtimeinfo");
+  }
+
+  /** Generates log file within certain path */
+  private void generateLogFile(LoggingContext loggingContext, String basePath, String filePath)
+      throws Exception {
+    long currentTimeMillis = System.currentTimeMillis();
+    long fromMillis = currentTimeMillis - TimeUnit.DAYS.toMillis(1);
+    CloseableIterator<LogEvent> logIter =
+        logReader.getLog(loggingContext, fromMillis, currentTimeMillis, FilterParser.parse(""));
+    AbstractChunkedLogProducer logsProducer = new TextChunkedLogProducer(logIter, logPattern, true);
+    ByteBuf chunk = logsProducer.nextChunk();
+    FileWriter file = new FileWriter(basePath + "/" + filePath);
+    while (chunk.capacity() > 0) {
+      String log = chunk.toString(StandardCharsets.UTF_8);
+      file.write(log);
+      chunk = logsProducer.nextChunk();
+    }
+    file.flush();
+    file.close();
```

Comment:
Use  tryWithResources (https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html) to ensure proepr cleanup in all the cases